pkg/products/grafana/reconciler.go

```diff
@@ -281,6 +282,49 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, client k8sclient.C
 
 	logrus.Info("Grafana Status: ", status)
 
+	prometheusNamespace := fmt.Sprintf("%smiddleware-monitoring-operator", global.NamespacePrefix)
+
+	url := fmt.Sprintf("http://%s.%s.svc:%s", "prometheus-service", prometheusNamespace, "9090")
```

Comment:
Would there ever be a need to make this URL configurable? Can't think of a reason off the top of my head but seeing hardcoded entries always makes me wary

Reply:
Yes, if the service name or port changed we'd end up creating the datasource without error but grafana wouldn't be able to access it. An alternative here would be to get the service, the namespace would still be hardcoded, at least the `middleware-monitoring-operator` and `service name` piece but it would be a little bit more robust. We could error if no service is found.

Reply:
It seems that `prometheus-service` is hardcoded in our monitoring checks so if it ever changed the alert should fire as part of release testing: https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/monitoring/prometheusRules.go#L270

I think we can leave this as is for now